includes/snmp.inc.php

```diff
@@ -177,6 +177,9 @@ function gen_snmp_cmd($cmd, $device, $oids, $options = null, $mib = null, $mibdi
     }
 
     $pollertarget = Device::pollerTarget($device);
```

Comment:
You should add this change to inside the `Device::pollerTarget()` function instead, and add a comment on why this is needed

Reply:
I intentionally didn't put it in `Device::pollerTarget()` because doing so would break its other callers (`device_is_up()`, `add_service()`, `includes/polling/unix-agent.inc.php`), which expect IPv6 literals *not* to be enclosed in brackets.  It's only `snmpget` & friends that require brackets, which is why I only put it here.

I've added a comment explaining why this is needed, and why only here.

Reply:
I think some other services need it also, like unix-agent and maybe pingcheck?

Reply:
PingCheck (`app/Jobs/PingCheck.php`) passes it to `fping`'s stdin, which expects it without brackets.  You're right about unix-agent though; I overlooked `fsockopen`'s semantics.  I moved this into `pollerTarget` and added an optional boolean parameter to control it.  How's that?